File: sdks/python/bolt_prediction_sdk/api/api_audit_log.py
"""
ApiAuditLog API 客户端
"""

from typing import Optional, Dict, Any, List, AsyncIterator
import json

from ..core.client import BaseAPIClient
from ..core.pagination import CursorPaginator
from ..models import *


class ApiAuditLogClient(BaseAPIClient):
    """ApiAuditLog API 客户端"""

    async def query_audit_logs_api_v1_auth_audit_logs_get(
        self,
        key_id: Optional[Any] = None,
        path: Optional[Any] = None,
        method: Optional[Any] = None,
        start_time: Optional[Any] = None,
        end_time: Optional[Any] = None,
        limit: Optional[int] = None,
        offset: Optional[int] = None
) -> ApiAuditLogListResponse:
        """
        查询API审计日志
        """

        response = await self._request(
            method="GET",
            path=f"/api/v1/api/v1/auth/audit-logs",
            params={
                "key_id": key_id,
                "path": path,
                "method": method,
                "start_time": start_time,
                "end_time": end_time,
                "limit": limit,
                "offset": offset,
            },
        )

        return response

File: sdks/typescript/src/api/apiAuditLog.ts
/** ApiAuditLog API 客户端 */

import { BaseAPIClient } from "../core/client";
import { CursorPaginator } from "../core/pagination";
import * as Models from "../models";

export class ApiAuditLogClient extends BaseAPIClient {

  /**
   * 查询API审计日志
   */
  async queryAuditLogsApiV1AuthAuditLogsGet(
    keyId?: any,
    path?: any,
    method?: any,
    startTime?: any,
    endTime?: any,
    limit?: number,
    offset?: number
  ): Promise<Models.ApiAuditLogListResponse> {
    const params: Record<string, any> = {};
    if (keyId !== undefined) params['key_id'] = keyId;
    if (path !== undefined) params['path'] = path;
    if (method !== undefined) params['method'] = method;
    if (startTime !== undefined) params['start_time'] = startTime;
    if (endTime !== undefined) params['end_time'] = endTime;
    if (limit !== undefined) params['limit'] = limit;
    if (offset !== undefined) params['offset'] = offset;

    return this._request(
      "GET",
      `/api/v1/api/v1/auth/audit-logs`,
      params,
      undefined
    );
  }
}